### DISCUSSION TITLE
Please list me reasons for this issue.

### DISCUSSION THREAD

**Discussion by ozgurerdogan** [psy0rz/zfs_autobackup#198] [General]
I am trying to create multi target snapshots. But time to time I am getting:
Cant find common snapshot with source, You probably need to delete the target dataset to fix this.

So can you please list me possible causes for this issue. Because everytime I have to start full sync. I am trying to find out for days..

**Comment by psy0rz** [comment]
please show the parameters you've used.

also the docs are recently updated for this:

https://github.com/psy0rz/zfs_autobackup/wiki#multiple-backups-of-the-same-data


Make sure your keep-source is big enough, so that one backup job doesnt delete common snapshot of the other one.

**Comment by ozgurerdogan** [comment]
Please check:

ssh sourceserver"zfs-autobackup -v --ssh-target root@target1 s1d8-backup D3 --keep-source=40 --keep-target=40 --strip-path=1 --clear-mountpoint --clear-refreservation --no-holds --snapshot-format %Y-%m-%d-%H:%M:%S --destroy-missing=10d --force --destroy-incompatible";

ssh sourceserver "zfs-autobackup -v --ssh-target root@target2 s1d8-backup D3 --keep-source=1000 --keep-target=40  --no-holds --strip-path=1 --clear-mountpoint --clear-refreservation  --snapshot-format %Y-%m-%d-%H:%M:%S --destroy-missing=1d --force --destroy-incompatible --rate 8M";

Should I set --keep-source=1000 to 40 also? Then what if one jop fails..

**Comment by psy0rz** [comment]
If you use a keep-source=40 in both cases, then you should be able to miss 40 backup jobs before the problem start.

How often the jobs are running?

To be on the safe side, i normally would advise against --force --destroy-missing and --destroy-incompatible.

**Comment by ozgurerdogan** [comment]
Cron is set to 0 2,8,10,12,14,16,18,20,22 * * *
So far it is running fine for two days.. Shall I still edit parameters?

**Comment by psy0rz** [comment]
both jobs run at these times?

If you run that often, you have the problem that if one job fails, the other job will delete the common snapshot after 40 runs. (which  is in 4 hours in your case)

Its better to use 2 backup names to prevent this. 

e.g. s1d8-backupA and s1d8-backupB for example.

or something like s1d8-backup-(targetname)

then you will never have these problems

**Comment by ozgurerdogan** [comment]
Ok I will move to two name version. But can I give two names to datasets? Those names are coming from dataset property name. zfs set autobackup:name1,name2 ?
